airbyte-ci/connectors/connector_ops/connector_ops/utils.py

```diff
@@ -285,35 +287,69 @@ def _get_type_and_name_from_technical_name(self) -> Tuple[str, str]:
         name = self.technical_name[len(_type) + 1 :]
         return _type, name
 
+    @property
+    def technical_name(self) -> str:
+        """
+        Return the technical name of the connector from the given relative_connector_path
+        e.g. source-google-sheets -> source-google-sheets or third-party/farosai/airbyte-pagerduty-source -> airbyte-pagerduty-source
+        """
+        return self.relative_connector_path.split("/")[-1]
+
     @property
     def name(self):
         return self._get_type_and_name_from_technical_name()[1]
 
     @property
     def connector_type(self) -> str:
-        return self._get_type_and_name_from_technical_name()[0]
+        return self.metadata["connectorType"] if self.metadata else None
 
     @property
-    def documentation_directory(self) -> Path:
+    def is_third_party(self) -> bool:
+        return THIRD_PARTY_GLOB in self.relative_connector_path
+
+    @property
+    def has_airbyte_docs(self) -> bool:
+        return (
+            self.metadata
+            and self.metadata.get("documentationUrl") is not None
+            and BASE_AIRBYTE_DOCS_URL in str(self.metadata.get("documentationUrl"))
+        )
+
+    @property
+    def local_connector_documentation_directory(self) -> Path:
         return Path(f"./docs/integrations/{self.connector_type}s")
 
+    @property
+    def relative_documentation_path_str(self) -> str:
+        documentation_url = self.metadata["documentationUrl"]
+        relative_documentation_path = documentation_url.replace(BASE_AIRBYTE_DOCS_URL, "")
+
+        # strip leading and trailing slashes
+        relative_documentation_path = relative_documentation_path.strip("/")
+
+        return f"./docs/{relative_documentation_path}"
+
     @property
     def documentation_file_path(self) -> Path:
-        readme_file_name = f"{self.name}.md"
-        return self.documentation_directory / readme_file_name
+        if not self.has_airbyte_docs:
+            return None
+
+        return Path(f"{self.relative_documentation_path_str}.md")
 
     @property
     def inapp_documentation_file_path(self) -> Path:
-        readme_file_name = f"{self.name}.inapp.md"
-        return self.documentation_directory / readme_file_name
+        if not self.has_airbyte_docs:
+            return None
+
+        return Path(f"{self.relative_documentation_path_str}.inapp.md")
```

Comment:
Should we raise an error when has_airbyte_docs is True but path does not exists?

Reply:
Im unsure. A hard error would prevent the QA checks from running on a list of connectors as it would fail if just one was missing a documentation file. Is that a large concern?